netket/abstract_variational_driver.py

```diff
@@ -168,14 +174,36 @@ def run(
             :step_size: Every how many steps should observables be logged to disk (default=1)
             :show_progress: If true displays a progress bar (default=True)
         """
+
+        # TODO Remove this deprecated code in v3.0
+        # manage deprecated where argument names are not specified, and
+        # prefix is passed as the first positional argument and the number
+        # of iterations as a second argument.
+        if type(n_iter) is str and type(output_prefix) is int:
+            # switch the two, and show a deprecation warning
+            tmp = n_iter
+            n_iter = output_prefix
+            output_prefix = tmp
+
+            warnings.warn(
+                "run(output_prefix, n_iter, **args) will be deprecated in version 3, use vmc.run(n_iter, output_prefix, **args) instead.",
+                PendingDeprecationWarning,
```

Comment:
We should use warnings consistently.  `PendingDeprecationWarning` is meant for things that may be deprecated in the future. Deprecation is not the same as removal. if we say "this will no longer work in 3.0, better update your code now", we are deprecating it _right now_ already. (That's why there is `DeprecationWarning`. if deprecation was the same as removal, having a warning for something that doesn't work would not make sense.)

However, both of these have the problem that they are not displayed to the user by default in may circumstances. Which means, normally people writing a script using NetKet will not see these warnings unless they specifically look for them. That's why we should use `FutureWarning` instead, to make sure those actually serve their purpose of alerting people that something will go away.

Reply:
Thanks for the explanation. I'm not very knowledgeable of deprecation warnings in python.
If you guys agree with the syntax I'll update them (and also those in `nk.variational.vmc` which I think also suffer from the same problem).

Reply:
I agree with the syntax, indeed it makes more sense to switch the order of arguments... I am just a bit unsure about the role of output_prefix in the new syntax... (also I would maybe switch its position with obs?) 

I mean, it seems to me that output_prefix, one one hand, is redundant because the logger should already know about where to write things? 
On the other hand, it's nice to have a quick way to fix the logger to output on a simple filename. Maybe the right solution would be to just remove output_prefix, check if logger gets a string and then that is interpreted as writing json to that file name prefix?

Reply:
@gcarleo Merging prefix and logger arguments in this way sounds good to me as well.